src/DynamoCore/Extensions/ExtensionLoader.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Dynamo.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml;
+
+namespace Dynamo.Extensions
+{
+    public class ExtensionLoader: IExtensionLoader
+    {
+        private IExtension Load(ExtensionDefinition extension)
+        {
+            try
+            {
+                var assembly = Assembly.Load(extension.AssemblyName);
+                var result = assembly.CreateInstance(extension.TypeName) as IExtension;
+                return result;
+            }
+            catch
+            {
+                return null;
+            }
```

Comment:
Let's make `ExtensionLoader` implement `ILogSource`.  When it fails to load an extension, let's log the appropriate failure.